### ISSUE TITLE
[clang-tidy] False warning

### ISSUE THREAD

**Issue by shrek1402** [llvm/llvm-project#53621]
`.clang-tidy` file:

```
Checks: >
  clang-diagnostic-*,
  clang-analyzer-*,
  bugprone*,
  clang-analyzer*,
  cppcoreguidelines*,
  google*,
  hicpp*,
  llvm-*,
  hicpp-*,
  modernize-use*,
  performance*,
  readability*,
  cert*,
  openmp*,
  -cert-err09*,
  -modernize-use-t*,
  -llvm-header-guard,
  -misc-non-private-member-variables-in-classes,
  -readability-redundant-access-specifiers,
  -bugprone-macro-parentheses,
  -cppcoreguidelines-prefer-member-initializer,
  -bugprone-easily-swappable-parameters,
  -readability-convert-member-functions-to-static,
  -cppcoreguidelines-pro-bounds-constant-array-index,
  -cert-env33-c,
  -readability-function-cognitive-complexity,
  -cppcoreguidelines-macro-usage
CheckOptions:
  - key:   readability-identifier-naming.AbstractClassCase
    value: 'lower_case'
  - key:   readability-identifier-naming.AbstractClassCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ClassCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ClassConstantCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ClassMemberCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ClassMethodCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ConstantCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ConstantMemberCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ConstantParameterCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ConstantPointerParameterCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ConstexprFunctionCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ConstexprMethodCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ConstexprVariableCase
    value: 'lower_case'
  - key:   readability-identifier-naming.EnumCase
    value: 'lower_case'
  - key:   readability-identifier-naming.EnumConstantCase
    value: 'lower_case'
  - key:   readability-identifier-naming.FunctionCase
    value: 'lower_case'
  - key:   readability-identifier-naming.GlobalConstantCase
    value: 'lower_case'
  - key:   readability-identifier-naming.GlobalConstantPointerCase
    value: 'lower_case'
  - key:   readability-identifier-naming.GlobalFunctionCase
    value: 'lower_case'
  - key:   readability-identifier-naming.GlobalPointerCase
    value: 'lower_case'
  - key:   readability-identifier-naming.GlobalVariableCase
    value: 'lower_case'
  - key:   readability-identifier-naming.InlineNamespaceCase
    value: 'lower_case'
  - key:   readability-identifier-naming.LocalConstantCase
    value: 'lower_case'
  - key:   readability-identifier-naming.LocalConstantPointerCase
    value: 'lower_case'
  - key:   readability-identifier-naming.LocalPointerCase
    value: 'lower_case'
  - key:   readability-identifier-naming.LocalVariableCase
    value: 'lower_case'
  - key:   readability-identifier-naming.MacroDefinitionCase
    value: 'UPPER_CASE'
  - key:   readability-identifier-naming.MemberCase
    value: 'lower_case'
  - key:   readability-identifier-naming.MethodCase
    value: 'lower_case'
  - key:   readability-identifier-naming.NamespaceCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ParameterCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ParameterPackCase
    value: 'lower_case'
  - key:   readability-identifier-naming.PointerParameterCase
    value: 'lower_case'
  - key:   readability-identifier-naming.PrivateMemberCase
    value: 'lower_case'
  - key:   readability-identifier-naming.PrivateMemberPrefix
    value: '_'
  - key:   readability-identifier-naming.PrivateMethodCase
    value: 'lower_case'
  - key:   readability-identifier-naming.PrivateMethodPrefix
    value: '_'
  - key:   readability-identifier-naming.ProtectedMemberCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ProtectedMemberPrefix
    value: '_'
  - key:   readability-identifier-naming.ProtectedMethodCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ProtectedMethodPrefix
    value: '_'
  - key:   readability-identifier-naming.PublicMemberCase
    value: 'lower_case'
  - key:   readability-identifier-naming.PublicMethodCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ScopedEnumConstantCase
    value: 'lower_case'
  - key:   readability-identifier-naming.StaticConstantCase
    value: 'lower_case'
  - key:   readability-identifier-naming.StaticVariableCase
    value: 'lower_case'
  - key:   readability-identifier-naming.StructCase
    value: 'lower_case'
  - key:   readability-identifier-naming.TemplateParameterCase
    value: 'CamelCase'
  - key:   readability-identifier-naming.TemplateTemplateParameterCase
    value: 'CamelCase'
  - key:   readability-identifier-naming.TypeAliasCase
    value: 'lower_case'
  - key:   readability-identifier-naming.TypedefCase
    value: 'lower_case'
  - key:   readability-identifier-naming.TypeTemplateParameterCase
    value: 'CamelCase'
  - key:   readability-identifier-naming.UnionCase
    value: 'lower_case'
  - key:   readability-identifier-naming.ValueTemplateParameterCase
    value: 'CamelCase'
  - key:   readability-identifier-naming.VariableCase
    value: 'lower_case'
  - key:   readability-identifier-naming.VirtualMethodCase
    value: 'lower_case'
WarningsAsErrors: ''
HeaderFilterRegex: '.*'
AnalyzeTemporaryDtors: false
FormatStyle: 'file'
```

code:

```
struct in_brackets {
  private:
    template<class TyHead, class... Ty>
    static std::wstring _body(TyHead head, Ty... values) {
    }
}
```

warning:
`warning: invalid case style for class method '_body' [readability-identifier-naming]`

But '_body' - valid case. It's warning appears only with 'static' functions.

**Comment by njames93** [comment]
Static class and struct members and methods are controlled by the Class(Member|Method) options. So you'll need to set ClassMethodPrefix to '_' to get the style you want
